RxJavaTemplate/app/src/main/java/co/nimblehq/rxjava/ui/screens/home/HomeViewModel.kt

```diff
@@ -42,11 +43,15 @@ class HomeViewModel @ViewModelInject constructor(
     }
 
     private fun fetchApi() {
+        IdlingResource.increment()
         getExampleDataUseCase
             .execute(Unit)
             .doShowLoading()
             .subscribeBy(
-                onSuccess = _data::onNext,
+                onSuccess = {
+                    _data.onNext(it)
+                    IdlingResource.decrement()
+                },
```

Comment:
why do we need to add `IdlingResource` functions here? Shouldn't it be called in test?

Reply:
also a ViewModel is usually containing only the real resources, when you want to run UI test or integration test with devices, we shouldn't duplicate what we already had in Unit Test, so you might actually want to bring these part to the View (Fragment) instead of this ViewModel logic code.

Reply:
Updated 🙏🏼